lib/rust.h

```diff
@@ -1,14 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0 */
-#ifndef _LIB_RUST_H
-#define _LIB_RUST_H
-
-#ifdef CONFIG_RUST
-char *rust_fmt_argument(char* buf, char* end, void *ptr);
-#else
-static inline char *rust_fmt_argument(char* buf, char* end, void *ptr)
-{
-	return NULL;
-}
-#endif
-
-#endif /* _LIB_RUST_H */
```

Comment:
I thought this is added after a feedback.

Reply:
Yeah, indeed. That feedback was from Greg to avoid `#ifdef`s, but now that one of the maintainers of this code suggested to use `IS_ENABLED`, I removed the header again and I went with a mix of Petr's and Rasmus' suggestions, which is also closer to other cases in the code (e.g. `CONFIG_OF`).